### PR TITLE
Fix a race condition in the marc export code which resulted in double…

### PR DESCRIPTION
**Pull request by jannistsiroyannis**
… exports of the same record.

### CODE REVIEW

**Review by kaipoykio — APPROVED** [review]
Nice!

**Review by andersju — APPROVED** [review]
LGTM!

File: marc_export/src/main/java/whelk/export/marc/ProfileExport.java [hunk 1]
```diff
@@ -357,9 +357,11 @@ private void exportDocument(Document document, ExportProfile profile, MarcRecord
...
         String profileName = profile.getProperty("name", "unknown");
         String systemId = document.getShortId();
-        if (exportedIDs.contains(systemId))
-            return;
-        exportedIDs.add(systemId);
+        synchronized (exportedIDs) {
+            if (exportedIDs.contains(systemId))
+                return;
+            exportedIDs.add(systemId);
+        }
```
Comment (olovy) [inline marc_export/src/main/java/whelk/export/marc/ProfileExport.java:364]: I suggest keeping the concurrent set and:
```suggestion
        if (!exportedIDs.add(systemId)) {
            return;
        }
```

Comment (jannistsiroyannis) [reply marc_export/src/main/java/whelk/export/marc/ProfileExport.java:364]: Hm. While also correct (ofc), I like the original one better. I think it's easier to read and understand. Yes, its subjective, I guess.

**Pull request merged**